pkg/server/container_create.go

```diff
@@ -597,7 +597,12 @@ func setOCIDevicesPrivileged(g *generate.Generator) error {
 // addOCIBindMounts adds bind mounts.
 func (c *criService) addOCIBindMounts(g *generate.Generator, mounts []*runtime.Mount, mountLabel string) error {
 	// Mount cgroup into the container as readonly, which inherits docker's behavior.
-	g.AddCgroupsMount("ro") // nolint: errcheck
+	g.AddMount(runtimespec.Mount{
+		Source:      "cgroup",
+		Destination: "/sys/fs/cgroup",
+		Type:        "cgroup",
+		Options:     []string{"nosuid", "noexec", "nodev", "relatime", "ro"},
```

Comment:
pretty big delta here.. do you have a link to this being the new expected default behavior options for cgroup? 

Will need to modify the above comment to reflect the additional changes, eg. and with nosuid to block the operation of suid and sgid bits (filesystem cannot contain set userid files), and...

Reply:
Actually, I just copied what the previous implementation of `g.AddCgroupsMount("ro")` was actually doing... So this should be a no-op. I don't see how the comment is not relevant, since it's not inaccurate to say the cgroup fs is being mounted read-only...

Reply:
Just took a look at the old code sitting on my master .. it looks to me like it would have returned early with nil? Might have been a coding error..
```
// AddCgroupsMount adds a cgroup mount into g.spec.Mounts.
func (g *Generator) AddCgroupsMount(mountCgroupOption string) error {
	switch mountCgroupOption {
	case "ro":
	case "rw":
	case "no":
		return nil
	default:
		return fmt.Errorf("--mount-cgroups should be one of (ro,rw,no)")
	}

	mnt := rspec.Mount{
		Destination: "/sys/fs/cgroup",
		Type:        "cgroup",
		Source:      "cgroup",
		Options:     []string{"nosuid", "noexec", "nodev", "relatime", mountCgroupOption},
	}
	g.initSpec()
	g.spec.Mounts = append(g.spec.Mounts, mnt)

	return nil
}
```

Reply:
Go is a [weird](https://stackoverflow.com/questions/40821855/do-go-switch-cases-fallthrough-or-not) language... :disappointed:

Reply:
:-) all good. So my read is that func was only able to check if the mount option was one of "ro,rw, or no" I presume no is for one of nosuid, nodev, noexec but we're not truncating so not sure what's up there. And for our case we just received a nil for the error. With the func skipping over the dunsel code by not having a break or continue but having a default. So we probably don't want to change up the options we were using unless there is a valid reason to change them.

Reply:
yeah nvm..